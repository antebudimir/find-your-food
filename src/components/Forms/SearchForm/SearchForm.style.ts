import styled from 'styled-components';

export const FormContainer = styled.form`
	padding-block: ${(props) => props.theme.spacing.primary};
`;
