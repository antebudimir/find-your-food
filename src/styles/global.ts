import { createGlobalStyle } from 'styled-components';
import RalewayRegular from 'fonts/Raleway-Regular.woff';
import RalewayBold from 'fonts/Raleway-Bold.woff';
import RalewayExtraBold from 'fonts/Raleway-ExtraBold.woff';

export const GlobalStyles = createGlobalStyle`
@font-face {
	font-family: 'Raleway Regular';
	src: url(${RalewayRegular}) format('woff');
	font-weight: normal;
	font-style: normal;
	font-display: swap;
}

@font-face {
	font-family: 'Raleway Bold';
	src: url(${RalewayBold}) format('woff');
	font-weight: bold;
	font-style: normal;
	font-display: swap;
}


@font-face {
    font-family: 'Raleway Extra Bold';
    src: url(${RalewayExtraBold}) format('woff');
    font-weight: bold;
    font-style: normal;
    font-display: swap;
}

* {
	outline: none;
	box-sizing: border-box;
	margin: 0;
	padding: 0;
	scroll-behavior: smooth;
}

html {
    font-family: 'Raleway Regular';
	word-wrap: break-word;
}

button,
input {
	border: none;
	display: block;
}

input,
select,
textarea,
button {
	color: inherit;
	font-size: inherit;
}

input,
select,
textarea {
	font-family: inherit;
}

h1 {
    font-family: 'Raleway Extra Bold';
}

h2,
button,
a {
	font-family: 'Raleway Bold';
}

ul,
li {
	list-style: none;
}

a {
	text-decoration: none;
}

img {
	display: block;
	max-width: 100%;
}
`;
