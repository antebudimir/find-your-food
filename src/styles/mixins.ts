import { css } from 'styled-components';

interface FlexLayoutProps {
	direction?: string;
	wrap?: string;
}

interface FlexAlignmentProps {
	justify?: string;
	align?: string;
}

export const FlexLayout = ({ direction, wrap }: FlexLayoutProps) => css`
	display: flex;
	flex-flow: ${direction} ${wrap};
`;

export const FlexAlignment = ({ justify, align }: FlexAlignmentProps) => css`
	justify-content: ${justify};
	align-items: ${align};
`;

export const hiddenVisually = css`
	position: absolute;
	clip: rect(0 0 0 0);
	clip-path: inset(50%);
	overflow: hidden;
	width: 1px;
	height: 1px;
	white-space: nowrap;
`;

export const sticky = css`
	display: flex;
	flex-flow: column nowrap;
	min-height: 89.6vh;
`;
