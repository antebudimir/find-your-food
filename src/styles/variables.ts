export const variables = {
	borderRadius: {
		primary: '0.125rem',
	},
	outline: {
		primary: '1px',
	},
	shadow: {
		primary: '#3b3f20 0px 2px 5px 0px',
		secondary: '#d45464 0px 2px 5px 0px',
		tertiary: '5px 5px 20px rgba(0, 0, 0, 0.7)',
		quaternary: '5px 5px 5px rgba(0, 0, 0, 0.7)',
	},
	sizing: {
		inputDesktopWidth: '28rem',
		summaryLinkWidth: '18.9rem', // coresponds to the width of the image
		summaryContainerWidth: '28rem',
		mainContainerWidth: '60rem',
	},
	spacing: {
		primary: '1rem',
		secondary: '0.5rem 1rem',
		tertiary: '2rem',
		quaternary: '1.5rem',
	},
	color: {
		primary: '#d45464',
		primaryLighter: '#ff6478',
		secondary: ' #ffffff',
		tertiary: '#3b3f20',
	},
	fontSize: {
		primary: '1rem',
		secondary: '1.3rem',
		tertiary: '1.5rem',
		quaternary: '2rem',
		quinary: '120%',
	},
	timing: {
		primary: '200ms',
	},
	lineHeight: {
		primary: '1.5',
		secondary: '2',
	},
};
