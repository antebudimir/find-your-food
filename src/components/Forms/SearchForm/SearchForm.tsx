import { FormContainer } from './SearchForm.style';
import { useState } from 'react';
import { useNavigate } from 'react-router';
import InputGroup, {
	InputType,
} from 'components/Elements/InputGroup/InputGroup';
import Button, { ButtonType } from 'components/Elements/Button/Button';

interface SearchFormProps {
	setSearchTerm: (searchTerm: string) => void;
}

const SearchForm = ({ setSearchTerm }: SearchFormProps) => {
	const [input, setInput] = useState<string>('');
	let navigate = useNavigate();

	const onSubmitHandler = (e: React.FormEvent<HTMLFormElement>) => {
		e.preventDefault();

		if (input !== '') {
			setSearchTerm(input);
			navigate('/recipes');
			setInput('');
		}
	};

	return (
		<FormContainer onSubmit={onSubmitHandler}>
			<InputGroup
				name="search-input"
				label="Enter an ingredient"
				type={InputType.search}
				placeholder="Enter an ingredient"
				value={input}
				setInput={setInput}
			/>

			<Button
				buttonType={ButtonType.submit}
				buttonTitle="Search for meals"
				buttonText="Search"
			/>
		</FormContainer>
	);
};

export default SearchForm;
