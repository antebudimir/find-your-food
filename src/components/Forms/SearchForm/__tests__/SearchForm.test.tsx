import { ThemeProvider } from 'styled-components';
import { variables } from 'styles/variables';
import '@testing-library/jest-dom';
import { render, screen, rerender } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import SearchForm, { SearchFormProps } from '../SearchForm';

const props: SearchFormProps = {
	setSearchTerm: jest.fn(),
};

const renderSearchForm = (props: SearchFormProps) => {
	const history = createMemoryHistory();

	const { asFragment } = render(
		<Router location={history.location} navigator={history}>
			<ThemeProvider theme={variables}>
				<SearchForm {...props} />
			</ThemeProvider>
		</Router>,
	);

	return { fragment: asFragment(), history };
};

// Render
describe('<SearchForm /> component', () => {
	test('Input field and button render correctly', () => {
		renderSearchForm(props);

		const input = screen.getByPlaceholderText('Enter an ingredient');
		expect(input).toBeVisible();
		expect(input).toHaveAttribute('type', 'search');

		const searchButton = screen.getByTitle('Search for meals');
		expect(searchButton).toBeVisible();
		expect(searchButton).toHaveAttribute('type', 'submit');
	});

	test('if the Component matches snapshot', () => {
		const { fragment } = renderSearchForm(props);
		expect(fragment).toMatchSnapshot();
	});
});

// Input field
describe('Input field', () => {
	test('Value updates on change', () => {
		renderSearchForm(props);

		const searchInput = screen.getByPlaceholderText('Enter an ingredient');
		userEvent.type(searchInput, 'Test search');
		expect(searchInput).toHaveValue('Test search');
	});
});

// Search button
describe('Search button', () => {
	it('should render with a red background', () => {
		renderSearchForm(props);

		const searchButton = screen.getByTitle('Search for meals');
		expect(searchButton).toHaveStyle('background-color: #d45464');
	});

	describe('Empty input field', () => {
		test('Does not submit the form', () => {
			const { history } = renderSearchForm(props),
				baseURL = '/';

			const searchButton = screen.getByTitle('Search for meals');
			userEvent.click(searchButton);
			expect(props.setSearchTerm).not.toHaveBeenCalled();
			expect(history.location.pathname).not.toMatch(baseURL + 'recipes');
		});
	});

	describe('User typed a query in the input field', () => {
		test('Submits the form, resets the input field and navigates to the recipes route', () => {
			const { history } = renderSearchForm(props),
				baseURL = '/';

			const searchInput = screen.getByPlaceholderText('Enter an ingredient');
			userEvent.type(searchInput, 'Recipe search');

			const searchButton = screen.getByTitle('Search for meals');
			userEvent.click(searchButton);
			expect(props.setSearchTerm).toHaveBeenCalled();
			expect(searchInput).toBeEmptyDOMElement();
			expect(history.location.pathname).toMatch(baseURL + 'recipes');
		});
	});
});
