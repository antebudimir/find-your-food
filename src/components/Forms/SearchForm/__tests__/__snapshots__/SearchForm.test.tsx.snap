// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> component if the Component matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="sc-bczRLJ cYufwz"
  >
    <div
      class="sc-gsnTZi"
    >
      <label
        class="sc-dkzDqf jMnMkt"
        for="search-input"
      >
        Enter an ingredient
      </label>
      <input
        class="sc-hKMtZM jwztwx"
        id="search-input"
        name="search-input"
        placeholder="Enter an ingredient"
        required=""
        type="search"
        value=""
      />
    </div>
    <button
      class="sc-eCYdqJ dKmmch"
      title="Search for meals"
      type="submit"
      value=""
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
